snapcraft/internal/sources/_git.py

```diff
@@ -90,3 +90,25 @@ def pull(self):
             self._pull_existing()
         else:
             self._clone_new()
+        self.assets = self._get_source_details()
+
+    def _get_source_details(self):
```

Comment:
This is missing `source-checksum`. I also don't understand how this will work if done in `__init__` where the source hasn't been pulled to `source_dir` yet.